test/api/v3/integration/groups/POST-groups_groupId_join.js

```diff
@@ -0,0 +1,63 @@
+import {
+  generateUser,
+} from '../../../../helpers/api-integration.helper';
+
+describe('POST /group/:groupId/join', () => {
+  let user;
+
+  beforeEach(() => {
+    return generateUser().then((generatedUser) => {
+      user = generatedUser;
+    });
+  });
+
+  context('Guilds', () => {
+    it('inviting to guild doesn\'t give basilist to inviter', () => {
+      let groupName = 'Test Guild';
+      let groupType = 'guild';
+
+      return generateUser({balance: 1, 'items.quests.basilist': 0 }).then((generatedUser) => {
+        user = generatedUser;
+        return user.post('/groups', {
+          name: groupName,
+          type: groupType
+        });
+      })
+      .then((guild) => {
+        return generateUser({ 'invitations.guilds': [{ id: guild._id}]}).then((generatedUser) => {
+          return generatedUser.post('/groups/' + guild._id + '/join');

```

Comment:
I [wrote a helper](https://github.com/HabitRPG/habitrpg/blob/02f4a841463c87159041c2e66b2faec60caa140d/test/helpers/api-integration.helper.js#L132-L212) (for v2 routes) to generate a group with members and invitees.

Once the invite routes are done, that should be used instead. Until them, I'd be fine with adding a todo comment to impliment it.